docs/fastdds/property_policies/non_consolidated_qos.rst

```diff
@@ -11,11 +11,63 @@ Before consolidating a new QoS Policy, it is usually set using this generic QoS
+
+DataWriters periodically announce the state of their queue by means of a heartbeat.
+Upon reception of the heartbeat, DataReaders will request the DataWriter to send the samples they want to process.
+Consequently, the publishing rate can be tuned setting the heartbeat period accordingly.
+
+.. list-table::
+   :header-rows: 1
+   :align: left
+
+   * - PropertyPolicyQos name
+     - PropertyPolicyQos value
+     - Default value
+   * - ``"fastdds.push_mode"``
+     - ``"true"``/``"false"``
+     - ``"true"``
+
++----------------------------------------------------------------------------------------------------------------------+
+| **C++**                                                                                                              |
++----------------------------------------------------------------------------------------------------------------------+
+| .. literalinclude:: /../code/DDSCodeTester.cpp                                                                       |
+|    :language: c++                                                                                                    |
+|    :start-after: // PULL_MODE_DATAWRITER                                                                             |
+|    :end-before: //!--                                                                                                |
+|    :dedent: 8                                                                                                        |
++----------------------------------------------------------------------------------------------------------------------+
+| **XML**                                                                                                              |
++----------------------------------------------------------------------------------------------------------------------+
+| .. literalinclude:: /../code/XMLTester.xml                                                                           |
+|    :language: xml                                                                                                    |
+|    :start-after: <!-->PULL_MODE_DATAWRITER<-->                                                                       |
+|    :end-before: <!--><-->                                                                                            |
++----------------------------------------------------------------------------------------------------------------------+
+
+.. note::
+    Communication to readers running on the same process (:ref:`intraprocess-delivery`) will always use ``push mode``
 
 .. warning::
-    This section is still under work.
+    It is inconsistent to enable the ``pull mode`` and also set the |ReliabilityQosPolicyKind-api| to
+    |BEST_EFFORT_RELIABILITY_QOS-api|.
+
+    It is inconsistent to enable the ``pull mode`` and also set the |WriterTimes::heartbeatPeriod-api| to
```

Comment:
I'd rather have either two warnings or an unordered list here. It looks funny on the HTML

Reply:
Done on 07f2be050dc0fbb63c75b52b847d793a52d53bec